Fmendezc98 create start page

fmendezc98 opened the pull request.

fmendezc98 committed:
Add primitive start page and route

fmendezc98 committed:
Add functional material-ui button and shared.js file

Added material-ui as dependency in package.js. File shared.js is currently not being used but was created for future development.

fmendezc98 committed:
Create majority of pages for basic user flows

Majority of pages are not yet populated

btheteach commented on src/client/main.js:
```diff
@@ -4,6 +4,9 @@ import { render } from "react-dom";
+
 
 const foo = () => (
 	<div>Hello Landing</div>
```
Why do we need this here?

fmendezc98 replied:
Deleted dummy method foo

btheteach commented on src/client/main.js:
```diff
@@ -19,7 +22,9 @@ class App extends Component {
 			<BrowserRouter>
```
A theme will be material-ui theme will be applied here in future branches

fmendezc98 replied:
Let's do this together on Thursday

btheteach commented on src/client/components/start.js:
```diff
@@ -0,0 +1,36 @@
+  grid-area: main;
+`;
+
+export class Start extends Component {
```
Start is confusing, is this where you end up once you login? Does it take over functioning as the landing page? Why do we need a start and a landing page?

fmendezc98 replied:
Landing was just a dummy page. It will be deleted in the next commit.

btheteach replied:
All good

btheteach commented on src/client/components/start.js:
```diff
@@ -0,0 +1,36 @@
+                    <Button component={Link} to="/report-category" variant="contained" color="primary">
+                        Report
+                    </Button>
+                    <Button component={Link} to="/" variant="contained" color="primary">
```
Does clicking Browse to get "/" path just render complaints launched previously or is that back to the site originally opened?

fmendezc98 replied:
Changed the path to "browse.js" for Browse button

btheteach commented on src/client/components/shared.js:
```diff
@@ -0,0 +1,16 @@
+/*************************************************************************/
+
+export const GenericButton = styled.button`
+  max-width: 200px;
```
Instead of using px convert to either strictly rem or em. Change globally

fmendezc98 replied:
Let's discuss this on Thursday also. Unsure about whether we even need a shared.js file since we might be fine using material-ui components

btheteach commented on src/client/components/shared.js:
```diff
@@ -0,0 +1,16 @@
+  background: #6495ed;
+  border: none;
+  border-radius: 5px;
+  line-height: 2em;
```
We need to decide are we going to use em's or rem's globally

btheteach commented on src/client/components/report_category.js:
```diff
@@ -0,0 +1,52 @@
+                        <MenuItem value="">
+                            <em>None</em>
+                        </MenuItem>
+                        <MenuItem value={10}>Ten</MenuItem>
```
What is the purpose of these menu items?

btheteach commented on src/client/components/report_category.js:
```diff
@@ -0,0 +1,52 @@
+import React, { Component } from "react";
+import styled from "styled-components";
+import Select from '@material-ui/core/Select';
+import { Link } from 'react-router-dom';
```
If we don't use the statement make sure to do clean up so we don;t have an overly large js file

fmendezc98 committed:
Addressed majority of Github comments from 3 original commits

Also edited report-category.js but will need to change it before our meeting on Thursday
